### PR TITLE
Update csi-addons to v0.10.0

### PR DESCRIPTION
**Pull request by nirs**
Required to consume the new Validated condition.

Include external snapshotter fix from #1565 (many tests fail in main).

### CODE REVIEW

**Review by BenamarMk — APPROVED** [review]

File: go.mod
Comment (nirs) [inline go.mod:1]: @ShyamsundarR posted a better fix here:
https://github.com/RamenDR/ramen/pull/1565/commits/a2a956f8814647ac343e492c0ab7e0c20ff92096

File: test/addons/csi-addons/kustomization.yaml [hunk 1]
```diff
@@ -3,10 +3,10 @@
 
 ---
 resources:
-  - https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/v0.9.1/deploy/controller/crds.yaml
-  - https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/v0.9.1/deploy/controller/rbac.yaml
-  - https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/v0.9.1/deploy/controller/setup-controller.yaml
+  - https://github.com/csi-addons/kubernetes-csi-addons/releases/download/v0.10.0/crds.yaml
```
Comment (ShyamsundarR) [inline test/addons/csi-addons/kustomization.yaml:6]: If we are not pointing to a kustomize file at the URL then we could stay with raw content instead? This does not reduce lines to update here as versions change.

Comment (nirs) [reply test/addons/csi-addons/kustomization.yaml:6]: But download url is much nicer than raw url - shorter, and more important, use something which is officially part of the release. The result should be the same, github probably redirect to the same file.

Comment (ShyamsundarR) [reply test/addons/csi-addons/kustomization.yaml:6]: Ah ok you are pointing to the download, I thought you were again going with the git clone

Comment (nirs) [reply test/addons/csi-addons/kustomization.yaml:6]: I wish we had kustomization upstream so we can do the git clone :-)

**Review by ELENAGER — APPROVED** [review]

**Pull request merged**